File: serverSpring/src/main/java/com/vyatsu/auctionComplex/entity/api/AuctionItem.java
package com.vyatsu.auctionComplex.entity.api;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.Instant;
import java.util.LinkedHashSet;
import java.util.Set;

@Entity
@Table(name = "auction_items")
public class AuctionItem {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    @Getter
    @Setter
    private Integer id;

    @Column(name = "title", length = 250)
    @Getter
    @Setter
    private String title;

    @Column(name = "description", length = 1000)
    @Getter
    @Setter
    private String description;

    @Column(name = "investment_size", nullable = false)
    @Getter
    @Setter
    private Double investmentSize;

    @Column(name = "location")
    @Getter
    @Setter
    private String location;

    @Column(name = "category")
    @Getter
    @Setter
    private String category;

    @Column(name = "start_date", nullable = false)
    @Getter
    @Setter
    private Instant startDate;

    @Column(name = "end_date")
    @Getter
    @Setter
    private Instant endDate;

    @ManyToOne()
    @JsonIgnore
    @JoinColumn(name = "winner_id")
    @Getter
    @Setter
    private User winner;

    @ManyToOne()
    @JsonIgnore
    @JoinColumn(name = "auctioneer_id", nullable = false)
    @Getter
    @Setter
    private User auctioneer;

    @Column(name = "is_valid")
    @Getter
    @Setter
    private String isValid;

    @OneToMany(mappedBy = "auctionItem")
    @JsonIgnore
    @Getter
    @Setter
    private Set<AuctionChat> auctionChats = new LinkedHashSet<>();

    @OneToMany(mappedBy = "auctionItem")
    @JsonIgnore
    @Getter
    @Setter
    private Set<Bid> bids = new LinkedHashSet<>();

}

File: adminWF/Classes/AuctionItem.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace auctionComplex.Classes
{
    [Table("auction_items")]
    internal class AuctionItem
    {
        [Key]
        [Column("id")]
        public int Id { get; set; }

        [Column("title"), MaxLength(250)]
        public string Title { get; set; }

        [Column("description"), MaxLength(1000)]
        public string Description { get; set; }

        [Column("investment_size")]
        public double InvestmentSize { get; set; }

        [Column("location"), MaxLength(500)]
        public string Location { get; set; }

        [Column("category"), MaxLength(100)]
        public string Category { get; set; }

        [Column("start_date")]
        public DateTime StartDate { get; set; } = DateTime.Now;

        [Column("end_date")]
        public DateTime? EndDate { get; set; }

        [ForeignKey("WinnerId")]
        public User Winner { get; set; }
        [Column("winner_id")]
        public int? WinnerId { get; set; }

        public User Auctioneer { get; set; }
        [Column("auctioneer_id")]
        public int AuctioneerId { get; set; }

        [Column("is_valid")]
        public string IsValid { get; set; } = "verification";

        public virtual ICollection<AuctionChat> AuctionChats { get; } = new List<AuctionChat>();
    }
}
